algorithms/registration-toolbox/include/registration-toolbox/alignment/pcl-alignment.h

```diff
@@ -123,48 +125,45 @@ void PclAlignment<T_alignment, T_point>::downSamplePointCloud(
   // point cloud fit into memory and otherwise skips downsampling, we have to
   // split the point cloud into multiple volumes that are small enough to fit.
 
-  size_t volume_level = 1;
+  std::uint64_t volume_level = 1;
 
   T_point min_point;
   T_point max_point;
   pcl::getMinMax3D(*cloud, min_point, max_point);
-  const std::int64_t dx = static_cast<std::int64_t>(
-                              (max_point.x - min_point.x) * inverse_grid_size) +
-                          1;
-  const std::int64_t dy = static_cast<std::int64_t>(
-                              (max_point.y - min_point.y) * inverse_grid_size) +
-                          1;
-  const std::int64_t dz = static_cast<std::int64_t>(
-                              (max_point.z - min_point.z) * inverse_grid_size) +
-                          1;
-  const std::int64_t dxdydz = dx * dy * dz;
+  const std::uint64_t dx =
+      std::max(std::ceil((max_point.x - min_point.x) * inverse_grid_size), 1.f);
+  const std::uint64_t dy =
+      std::max(std::ceil((max_point.y - min_point.y) * inverse_grid_size), 1.f);
+  const std::uint64_t dz =
+      std::max(std::ceil((max_point.z - min_point.z) * inverse_grid_size), 1.f);
+  const std::uint64_t dxdydz = dx * dy * dz;
   // Find amount of levels that we have to split the pointcloud into
   bool indices_fit_in_voxel_grid = false;
   while (!indices_fit_in_voxel_grid) {
     if ((dxdydz / (volume_level * volume_level * volume_level)) <
-        static_cast<std::int64_t>(std::numeric_limits<std::int32_t>::max())) {
+        static_cast<std::uint64_t>(std::numeric_limits<std::int32_t>::max())) {
```

Comment:
`std::numeric_limits<std::uint64_t>::max`?

Reply:
this is intentionally `std::numeric_limits<std::int32_t>::max()` since this is what pcl is checking for